site/src/xServices/updateCheck/updateCheckXService.ts

```diff
@@ -0,0 +1,103 @@
+import { assign, createMachine } from "xstate"
+import * as API from "../../api/api"
+import * as TypesGen from "../../api/typesGenerated"
+import { displayMsg } from "../../components/GlobalSnackbar/utils"
+
+export const Language = {
+  updateAvailable: "New version available",
+  updateAvailableMessage: (
+    version: string,
+    url: string,
+    upgrade_instructions_url: string,
+  ): string =>
+    `Coder ${version} is now available at ${url}. See ${upgrade_instructions_url} for information on how to upgrade.`,
+}
+
+export interface UpdateCheckContext {
+  getUpdateCheckError?: Error | unknown
+  updateCheck?: TypesGen.UpdateCheckResponse
+}
+
+export const updateCheckMachine = createMachine(
+  {
+    id: "updateCheckState",
+    predictableActionArguments: true,
+    tsTypes: {} as import("./updateCheckXService.typegen").Typegen0,
+    schema: {
+      context: {} as UpdateCheckContext,
+      services: {} as {
+        getUpdateCheck: {
+          data: TypesGen.UpdateCheckResponse
+        }
+      },
+    },
+    context: {
+      updateCheck: undefined,
+    },
+    initial: "gettingUpdateCheck",
+    states: {
+      gettingUpdateCheck: {
+        invoke: {
+          src: "getUpdateCheck",
+          id: "getUpdateCheck",
+          onDone: [
+            {
+              actions: [
+                "assignUpdateCheck",
+                "clearGetUpdateCheckError",
+                "notifyUpdateAvailable",
+              ],
+              target: "#updateCheckState.success",
+            },
+          ],
+          onError: [
+            {
+              actions: ["assignGetUpdateCheckError", "clearUpdateCheck"],
+              target: "#updateCheckState.failure",
+            },
+          ],
+        },
+      },
+      success: {
+        type: "final",
+      },
+      failure: {
+        type: "final",
+      },
+    },
+  },
+  {
+    services: {
+      getUpdateCheck: API.getUpdateCheck,
+    },
+    actions: {
+      assignUpdateCheck: assign({
+        updateCheck: (_, event) => event.data,
+      }),
+      clearUpdateCheck: assign((context: UpdateCheckContext) => ({
+        ...context,
+        updateCheck: undefined,
+      })),
+      assignGetUpdateCheckError: assign({
+        getUpdateCheckError: (_, event) => event.data,
+      }),
+      clearGetUpdateCheckError: assign((context: UpdateCheckContext) => ({
+        ...context,
+        getUpdateCheckError: undefined,
+      })),
+      notifyUpdateAvailable: (context: UpdateCheckContext) => {
+        if (context.updateCheck && !context.updateCheck.current) {
+          // TODO(mafredri): HTML formatting, persistance?
+          displayMsg(
+            Language.updateAvailable,
+            Language.updateAvailableMessage(
+              context.updateCheck.version,
+              context.updateCheck.url,
+              "https://coder.com/docs/coder-oss/latest/admin/upgrade",
+            ),
+          )
```

Comment:
We now have a new banner component which might be useful for this feature: `AlertBanner`. The [AlertBanner component](https://github.com/coder/coder/pull/4403) comes in three flavors, "warning", "error", and "info", and can be configured to be dismissible, or not. You're welcome to extend it and add your own flavor, too. It is persistent until dismissed. I think this design would easily incorporate links. There are other paths forward if this doesn't seem like a good solution - just let me know.

Reply:
@Kira-Pilot finally got back around to this PR, `AlertBanner` would indeed work but I’m wondering how to go about including links.

I know react has `dangerouslySetInnerHTML`, but it’s probably not best practice to use it. Are there any other ways I could render the links since the `text` property doesn’t parse HTML. One option could be to allow markdown to be passed/parsed, but perhaps that would be weird too.

Reply:
@mafredri It's been a while since I took a look at this; would you be open to discussing in Discord so I can make sure I understand the roadblock? I'm mostly available.

Reply:
@Kira-Pilot understandable. Since a picture is worth more than words, let's see.

This is what I want to achieve:

<img width="1441" alt="image" src="https://user-images.githubusercontent.com/147409/204520221-cc56f6ce-d336-4a81-a084-1267dab82935.png">

This is how I achieved it: [`ee967ae` (#4810)](https://github.com/coder/coder/pull/4810/commits/ee967ae321f2781f994990b582ec9b3aa6d9b9f0).

I'm wondering if this is acceptable or if there's a better way to do it.

Reply:
@mafredri I think that's a perfect way to do it!! Nice work!!